core/src/main/java/com/novoda/merlin/service/Ping.java

```diff
@@ -14,12 +15,16 @@
 
     public boolean doSynchronousPing() {
         MerlinLog.d("Pinging : " + hostAddress);
-        int responseCode = responseCodeFetcher.from(hostAddress);
-        MerlinLog.d("Got response : " + responseCode);
-        return isSuccess(responseCode);
-    }
+        try {
+            responseCodeFetcher.from(hostAddress);
+        } catch (RequestException e) {
+            if (e.causedByIO()) {
```

Comment:
Then you shouldn't let people pass an arbitrary url, because you don't know beforehand what's the HTTP code going to be returned. You should have ALWAYS an endpoint whose response is known beforehand so that if it changes you know something's going on not working.

So in that case the cause for success would be only 204 for the connectivity url provided by android. And that wouldn't be part of the dependencies of this class.